fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -491,16 +491,80 @@ struct CommitBatchContext {
 
 	std::unordered_map<uint16_t, Version> tpcvMap; // obtained from sequencer
 	std::set<uint16_t> writtenTLogs; // the set of tlog locations written to in the mutation.
-	std::set<Tag> writtenTags; // the set of tags written to in the mutation.
+	std::set<Tag> writtenTags; // final set tags written to in the batch
+	std::set<Tag> writtenTagsPreResolution; // tags written to in the batch not including any changes from the resolver.
+	bool hasMetadataMutation = false;
+	bool metadataMutationFromProxy = false;
 
 	CommitBatchContext(ProxyCommitData*, const std::vector<CommitTransactionRequest>*, const int);
 
 	void setupTraceBatch();
 
+	std::set<Tag> getWrittenTagsPreResolution();
+
 private:
 	void evaluateBatchSize();
 };
 
+std::set<Tag> CommitBatchContext::getWrittenTagsPreResolution() {
+	std::set<Tag> transactionTags;
+	std::vector<Tag> cacheVector = { cacheTag };
```

Comment:
A general comment here (or at some other appropriate place) to the effect that "this method computes both written tags and written tlogs and the tags computed here will be used for error checking and the actual tags used to update the version vector will be computed after the resolution step" will be helpful.